### PR TITLE
Example of how to setup a sidecar app

### PR DESCRIPTION
**Pull request by johnman**

### CODE REVIEW

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+	await initializeDom();
+	await initializePlatform();
+});
+
+/**
+ * Initialize the DOM elements.
+ */
+async function initializeDom(): Promise<void> {
+	loggingElement = document.querySelector("#logging");
+	const loggingContainer: HTMLDivElement | null = document.querySelector("#logging-container");
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: Just a comment really if you use generic `document.querySelector<HTMLDivElement>()` the loggingContainer is already the correct type

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+		loggingAddEntry(`There was an error while trying to initialize your platform. Error: ${getErrorMessage(error)}`);		
+	}
+}
+
+/**
+ * Return a message from an error.
+ * @param error The error to inspect and retrieve the message from
+ * @returns The error message
+ */
+function getErrorMessage(error: unknown): string {
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: We have a sort of standardized method like this, which we use all over the place https://github.com/built-on-openfin/workspace-starter/blob/bcf97bdf9b4cd8ca32ebed835b3483db6ffdef5d/how-to/workspace-platform-starter/client/src/framework/utils.ts#L112-L119

File: how-to.v2/use-a-sidecar-app-basic/platform/client/tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+		"rootDir": "./src",
+		"outDir": "build",
+		"skipLibCheck": true,
+		"strict": true,
+		"esModuleInterop": true,
+		"forceConsistentCasingInFileNames": true,
+		"moduleResolution": "node",
+		"types": ["./fin"]
+	},
+	"include": ["./src/**/*.ts", "fin.d.ts"]
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/platform/client/tsconfig.json]: not sure why fin.d.ts is in here, if anything the types on line 13 should say ["./types"] and the fin.d.ts file should be in a types folder, to match other repos

File: how-to.v2/use-a-sidecar-app-basic/platform/public/js/provider.bundle.js [hunk 4]
```diff
@@ -0,0 +1,228 @@
+/******/ (() => { // webpackBootstrap
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/platform/public/js/provider.bundle.js]: Should this file be excluded in the gitignore

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts [hunk 5]
```diff
@@ -0,0 +1,217 @@
...
+	if (error instanceof Error) return error.message
+	return String(error)
+}
+
+/**
+ * Create and assign a channel client
+ * @returns channel client
+ */
+async function createChannelClient(): Promise<ChannelClient> {
+	const channelName = "sidecar-app";
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: Can we make the channel names global consts and the same in the C# app, easier to spot them

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+				connectToSideCarBtn.disabled = false;
+			}
+			if(dispatchMessageToSideCarBtn !== null) {
+				dispatchMessageToSideCarBtn.disabled = true;
+			}
+			loggingAddEntry("Client has been disconnected from the SideCar App");
+		});
+
+		// received request over channel to display a message
+		channelClient.register('sidecar-app-client-subscriber', async (payload) => {
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: Same with naming here, can we make global const

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+}
+
+/** 
+ * Send an example message to the SideCar application.
+ */
+async function dispatchMessageToSideCarApp() {
+	try {
+		if(channelClient !== undefined) {
+			loggingAddEntry("Sending Message to SideCar App.");
+			let response = await channelClient.dispatch("sidecar-app-echo", "Message from platform to sidecar");
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: Global const

File: how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+            runtime.Disconnected += Runtime_Disconnected;
+
+            await runtime.ConnectAsync();
+
+            channels = runtime.GetService<IChannels>();
+
+            onAction("Creating Provider");
+            provider = channels.CreateProvider("sidecar-app");
+            onAction("Registering echo topic.");
+            provider.RegisterTopic<string, string>("sidecar-app-echo", (payload) =>
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs]: Move the topics to global consts

File: how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs [hunk 9]
```diff
@@ -0,0 +1,109 @@
...
+                onAction("Client Disconnected");
+            };
+            await provider.OpenAsync();
+        }
+
+        public async void BroadcastMessage(string message)
+        {
+            if(provider != null)
+            {
+                await provider.BroadcastAsync("sidecar-app-client-subscriber", message);
```
Comment (obany) [inline how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs]: Global const

**Review by obany — APPROVED** [review]
Approved, mentioned a few things but nothing that stops it working :)

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts (hunk 1 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: done

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts (hunk 2 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: copied (didn't copy the util file as I am trying to keep it small but copied the function)

File: how-to.v2/use-a-sidecar-app-basic/platform/client/tsconfig.json (hunk 3 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/platform/client/tsconfig.json]: updated

File: how-to.v2/use-a-sidecar-app-basic/platform/public/js/provider.bundle.js (hunk 4 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/platform/public/js/provider.bundle.js]: added to gitignore

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts (hunk 5 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: done

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts (hunk 6 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: done

File: how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts (hunk 7 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/platform/client/src/provider.ts]: done

File: how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs (hunk 8 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs]: done

File: how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs (hunk 9 above)
Comment (johnman) [reply how-to.v2/use-a-sidecar-app-basic/sidecar-app/Provider.cs]: done

**Pull request merged**